src/node_buffer.cc

```diff
@@ -499,6 +517,62 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness, T min, T max>
+void ReadInt64Generic(const FunctionCallbackInfo<Value>& args,
+                      const char* formatter) {
+  Environment* env = Environment::GetCurrent(args);
+
+  ARGS_THIS(args[0].As<Object>());
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset + sizeof(T), obj_length);
+
+  union NoAlias {
+    T val;
+    char bytes[sizeof(T)];
+  };
+
+  union NoAlias na;
+  const char* ptr = static_cast<const char*>(obj_data) + offset;
+  memcpy(na.bytes, ptr, sizeof(na.bytes));
+  if (endianness != GetEndianness())
+    Swizzle(na.bytes, sizeof(na.bytes));
+
+  if (na.val < min || na.val > max) {
```

Comment:
style nit: switch condition to the inverse (e.g. `if (!(na.val < min || na.val > max))`) so you can have an immediate one line return statement. this would remove the indentation for the lines within this if statement.